apps/passport-client/new-components/screens/Home/NewHomeScreen.tsx

```diff
@@ -220,15 +227,33 @@ export const NewHomeScreen = (): ReactElement => {
   const windowWidth = useWindowWidth();
   const self = useSelf();
   const navigate = useNavigate();
-  const isLoadedPCDs = useLoadedIssuedPCDs();
-
+  const isLoadedPCDs = useIsSyncSettled();
   const isInvalidUser = useUserForcedToLogout();
+  const location = useLocation();
+
   useEffect(() => {
     if (!self) {
       navigate("/login", { replace: true });
     }
   });
 
+  useLayoutEffect(() => {
+    // if we haven't loaded all pcds yet, dont process the prove request
+    if (!isLoadedPCDs) return;
```

Comment:
This might be a failure of understanding on my part, but I think what happens if I follow a link to the prove screen is this:

1) I see the home screen, and the home screen is interactive, meaning that I can click on stuff
2) Somewhere on the home screen, there's a visual indicator that my PCDs are loading
3) I don't see a modal anywhere
4) After some time, the modal appears

In the current flow, I see a loading spinner while I wait for the PCDs to load, and then I'm taken directly to the prove screen. In this flow, it's not clear that the prove screen is loading, because I see the home screen and no visual indicator that a modal is going to appear later. Sometimes PCDs take a long time to load (on bad wifi, or if the server is slow to respond), so users might just start clicking around on the home screen or assuming that the link is incorrect because they don't see the prove screen or any indicating that it's going to load.

Reply:
the loading indicator takes the whole screen, and you will immidiately open the modal when the pcds are loaded.